spec/features/consumer/shopping/checkout_stripe_spec.rb

```diff
@@ -79,4 +82,86 @@
       end
     end
   end
+
+  describe "using Stripe SCA" do
+    let!(:stripe_account) { create(:stripe_account, enterprise: distributor) }
+    let!(:stripe_sca_payment_method) {
+      create(:stripe_sca_payment_method, distributors: [distributor])
+    }
+    let!(:shipping_method) { create(:shipping_method) }
+    let(:payment_intent_id) { "pi_123" }
+    let(:payment_intent_response_mock) do
+      {
+        status: 200, body: JSON.generate(object: "payment_intent",
+                                         amount: 2000,
+                                         charges: { data: [{ id: "ch_1234", amount: 2000 }] })
+      }
+    end
+    let(:payment_intent_authorize_response_mock) do
+      {
+        status: 200, body: JSON.generate(id: payment_intent_id,
+                                         object: "payment_intent",
+                                         amount: 2000,
+                                         status: "requires_capture", last_payment_error: nil,
+                                         charges: { data: [{ id: "ch_1234", amount: 2000 }] })
+      }
+    end
+    let(:hubs_payment_method_response_mock) do
+      {
+        status: 200,
+        body: JSON.generate(id: "pm_456", customer: "cus_A123")
+      }
+    end
+
+    before do
+      allow(Stripe).to receive(:api_key) { "sk_test_12345" }
+      allow(Stripe).to receive(:publishable_key) { "pk_test_12345" }
+      Spree::Config.set(stripe_connect_enabled: true)
+
+      stub_request(:post, "https://api.stripe.com/v1/payment_intents")
+        .with(basic_auth: ["sk_test_12345", ""], body: /.*#{order.number}/)
```

Comment:
Some of these bits were copy-pasted from elsewhere. I'll have a look at refactoring this into something nicer and re-usable...

Reply:
ah, yes, good idea, some form of stripe_stubbing_helper that takes care of auth for example!